### PR TITLE
Add ClickBench to CI

### LINKED ISSUE #57708: A simple ClickBench check in the master branch.
We have detailed performance tests but don't have ClickBench in CI.

The idea is to make a check in the master branch that will attach a table with `web` disk plus cache (see https://github.com/ClickHouse/web-tables-demo), and run ClickBench queries.

It will take just a few minutes. We can run it on various machine types indiscriminately and with only the new version and collect the logs in our CI cluster.

As a side effect, we will have a hardware benchmark for free.

This check will be green regardless of the performance.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Run ClickBench for every commit. This closes https://github.com/ClickHouse/ClickHouse/issues/57708.

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 3ac3a06561f8f98fd3f38b9048a5b1a44c263377 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57712/3ac3a06561f8f98fd3f38b9048a5b1a44c263377/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7153915683">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57712/3ac3a06561f8f98fd3f38b9048a5b1a44c263377/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/ci/clickbench.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        post_commit_status_to_file(
+            post_commit_path,
+            description,
+            state,
+            report_url,
+        )
+    else:
+        raise Exception(
+            f'Unknown post_commit_status option "{args.post_commit_status}"'
+        )
```
Comment (Felixoid) [inline tests/ci/clickbench.py:229]: This is from `bugfix`, completely unrelated

**Review by Felixoid — COMMENTED** [review]
Pretty good from the code perspective, although raises some questions

**Comment by alexey-milovidov** [comment]
50 minutes is too much, it should be faster. I target around 15 minutes.
The idea is to put it in `master` or `daily`, then collect the data and sit on it (draw the historical charts).

There is also a secondary idea - that it will test `web` + `cache` disks with all our mechanics, but we should already have normal tests for that.

**Comment by alexey-milovidov** [comment]
We should copy the data to another region, because currently there is cross-region traffic.

**Comment by Felixoid** [comment]
> 50 minutes is too much, it should be faster. I target around 15 minutes.

Right now it's 50 minutes for [AMD](https://github.com/ClickHouse/ClickHouse/actions/runs/7162022865/job/19502263880) and 90 minutes for [arm64](https://github.com/ClickHouse/ClickHouse/actions/runs/7162022865/job/19502262458), just in case

**Comment by alexey-milovidov** [comment]
Maybe the cache is ineffective:
```
Mon, 11 Dec 2023 17:31:28 GMT [0.004, 0.002, 0.002],
Mon, 11 Dec 2023 17:31:32 GMT [3.699, 0.025, 0.023],
Mon, 11 Dec 2023 17:31:37 GMT [4.958, 0.061, 0.060],
Mon, 11 Dec 2023 17:31:57 GMT [6.658, 6.216, 6.760],
Mon, 11 Dec 2023 17:32:19 GMT [7.237, 7.017, 7.098],
Mon, 11 Dec 2023 17:32:48 GMT [10.478, 9.633, 9.605],
Mon, 11 Dec 2023 17:32:53 GMT [1.689, 1.141, 1.140],
Mon, 11 Dec 2023 17:32:53 GMT [0.043, 0.023, 0.024],
Mon, 11 Dec 2023 17:33:38 GMT [15.358, 14.712, 14.578],
Mon, 11 Dec 2023 17:34:58 GMT [26.693, 26.880, 26.248],
Mon, 11 Dec 2023 17:35:18 GMT [7.275, 6.648, 6.466],
Mon, 11 Dec 2023 17:35:38 GMT [6.573, 6.535, 6.545],
Mon, 11 Dec 2023 17:36:08 GMT [10.132, 9.805, 9.565],
Mon, 11 Dec 2023 17:37:50 GMT [32.438, 35.113, 34.496],
Mon, 11 Dec 2023 17:39:05 GMT [25.362, 24.863, 24.460],
Mon, 11 Dec 2023 17:39:42 GMT [22.780, 7.227, 7.265],
Mon, 11 Dec 2023 17:41:47 GMT [41.223, 40.766, 42.259],
Mon, 11 Dec 2023 17:43:49 GMT [40.500, 40.585, 40.565],
Mon, 11 Dec 2023 17:48:02 GMT [86.149, 79.240, 87.296],
Mon, 11 Dec 2023 17:48:24 GMT [7.523, 7.520, 7.414],
Mon, 11 Dec 2023 17:52:57 GMT [89.561, 91.009, 91.438],
Mon, 11 Dec 2023 17:59:58 GMT [142.330, 139.004, 140.010],
Mon, 11 Dec 2023 18:12:22 GMT [247.677, 244.766, 250.880],
Mon, 11 Dec 2023 18:24:40 GMT [274.408, 227.694, 236.531],
Mon, 11 Dec 2023 18:26:38 GMT [38.703, 37.609, 40.975],
Mon, 11 Dec 2023 18:27:07 GMT [9.175, 9.574, 10.109],
Mon, 11 Dec 2023 18:29:03 GMT [38.887, 37.818, 39.065],
Mon, 11 Dec 2023 18:34:53 GMT [116.106, 117.003, 116.907],
Mon, 11 Dec 2023 18:39:48 GMT [97.911, 98.597, 97.951],
Mon, 11 Dec 2023 18:39:50 GMT [0.669, 0.693, 0.643],
Mon, 11 Dec 2023 18:42:23 GMT [51.463, 50.502, 50.914],
Mon, 11 Dec 2023 18:45:34 GMT [62.417, 63.626, 64.770],
Mon, 11 Dec 2023 18:50:32 GMT [99.354, 98.137, 99.816],
Mon, 11 Dec 2023 18:55:06 GMT [92.571, 90.106, 91.228],
Mon, 11 Dec 2023 18:59:40 GMT [92.571, 91.808, 89.880],
Mon, 11 Dec 2023 18:59:58 GMT [5.930, 5.543, 5.771],
Mon, 11 Dec 2023 19:00:06 GMT [2.753, 2.688, 2.738],
Mon, 11 Dec 2023 19:00:13 GMT [2.390, 1.718, 2.243],
Mon, 11 Dec 2023 19:00:21 GMT [2.651, 2.657, 2.637],
Mon, 11 Dec 2023 19:00:31 GMT [3.135, 3.287, 3.508],
Mon, 11 Dec 2023 19:00:38 GMT [2.254, 2.102, 2.186],
Mon, 11 Dec 2023 19:00:44 GMT [2.116, 2.204, 2.196],
Mon, 11 Dec 2023 19:00:51 GMT [2.272, 2.075, 2.101],
```

**Review by alesapin — COMMENTED** [review]
I don't understand the idea of this check. If we want to compare performance -- why not to add this as performance test?

**Comment by alexey-milovidov** [comment]
@alesapin, the Performance Test is the set of "unit" tests for performance - small tests to control the performance in isolated, clean cases. I want the result of an end-to-end ClickBench test every day. See also https://benchmark.clickhouse.com/versions/

**Comment by alexey-milovidov** [comment]
In the same way, I could use a Lambda in AWS that runs [fully automated, self-driving benchmark on another machine](https://github.com/ClickHouse/ClickBench/blob/main/clickhouse-web/cloud-init.sh), but I want to try our CI infrastructure.

**Comment by alesapin** [comment]
`> I want the result of an end-to-end ClickBench test every day`
I'm not against "every day", but you want 2 jobs each of 1 hour for every commit.

File: tests/ci/ci_config.py [hunk 2]
```diff
@@ -507,6 +509,11 @@ def __hash__(self) -> int:
         "successfully startup without any errors, crashes or sanitizer asserts",
         lambda x: x.startswith("Upgrade check ("),
     ),
+    CheckDescription(
+        "ClickBench",
+        "Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table",
+        lambda x: x.startswith("Upgrade check ("),
```
Comment (alesapin) [inline tests/ci/ci_config.py:515]:
```suggestion
        lambda x: x.startswith("ClickBench"),
```

**Comment by alexey-milovidov** [comment]
`Insert into cache is skipped due to insufficient disk space.`

**Comment by alexey-milovidov** [comment]
Interesting - we have 64 GB of RAM, but not enough space in /dev/shm/

**Comment by alexey-milovidov** [comment]
Fixed - not the cache should work.

File: tests/ci/clickbench.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def get_run_command(
     env_str = " ".join(envs)
 
     return (
-        f"docker run --volume={builds_path}:/package_folder "
+        f"docker run --shm-size=16g --volume={builds_path}:/package_folder "
```
Comment (Felixoid) [inline tests/ci/clickbench.py:56]: It should be something like `max(16, ram-x)` to not overgrow, and being able to run anywhere

Comment (alexey-milovidov) [reply tests/ci/clickbench.py:56]: The data size is around 13 GB.

**Review by alesapin — COMMENTED** [review]
Let's also setup hard timeout for this check for 20 minutes, just in case. I want it because this check is always green and nobody will look at the results. If we will break something it will start to take hours it could be super costly.

**Comment by alexey-milovidov** [comment]
Now, 6 minutes and 7 minutes, correspondingly.

**Comment by alexey-milovidov** [comment]
@alesapin I've added a timeout. Plus, it appears so lightweight and nice (less than 10 minutes) that I decided to add it to master instead of nightly.

**Comment by Felixoid** [comment]
You have added the timeout in an inappropriate place.

It should be in https://github.com/ClickHouse/ClickHouse/pull/57712/files#diff-50ce4c0891f4c52bb0a90b064b447f582d3187727c178c0fa39cdef8c8932abcR176

**Pull request merged**